ale/drivers/base.py

```diff
@@ -439,21 +697,42 @@ def sun_velocity(self):
         return [sun_state[3:6].tolist()]
 
     @property
-    def sensor_position(self):
-
-        if not hasattr(self, '_sensor_position'):
-            self._compute_ephemerides()
-        return self._sensor_position.tolist()
+    def _sensor_position(self):
+        if not hasattr(self, '_position'):
+            eph = []
+            current_et = self.starting_ephemeris_time
+            for i in range(self.number_of_ephemerides):
+                state, _ = spice.spkezr(self.spacecraft_name,
+                                        current_et,
+                                        self.reference_frame,
+                                        'NONE',
+                                        self.target_name,) # If this is the sensor, insufficient, if this is the spacecraft, it works? Huh?
+                eph.append(state[:3])
+                current_et += getattr(self, "dt_ephemeris", 0)
+            # By default, spice works in km
+            self._position = [e * 1000 for e in eph]
+        return self._position
 
     @property
-    def sensor_velocity(self):
-        if not hasattr(self, '_sensor_velocity'):
-            self._compute_ephemerides()
-        return self._sensor_velocity.tolist()
+    def _sensor_velocity(self):
+        if not hasattr(self, '_velocity'):
+            eph_rates = []
+            current_et = self.starting_ephemeris_time
+            for i in range(self.number_of_ephemerides):
+                state, _ = spice.spkezr(self.spacecraft_name,
+                                        current_et,
+                                        self.reference_frame,
+                                        'NONE',
+                                        self.target_name,) # If this is the sensor, insufficient, if this is the spacecraft, it works? Huh?
+                eph_rates.append(state[3:])
+                current_et += getattr(self, "dt_ephemeris", 0)
+            # By default, spice works in km
+            self._velocity = [e*1000 for e  in eph_rates]
+        return self._velocity
```

Comment:
Would it be worth saving both the position and velocity in these methods? spkezr will return both of these.

Same with _sensor_position above.

Reply:
The previous approach felt clunky and broke form. If more people think otherwise I can revert it.